### ISSUE TITLE
Crash on link click

### ISSUE THREAD

**Issue by aaustin** [BranchMetrics/unity-branch-deep-linking-attribution#2]
When the link is touched, the game is opened normally (by the intent-filter), but we receive an error when calling 'Branch.Init' method:

```
07-03 17:12:13.878: I/Unity(32438): AndroidJavaException: java.lang.NullPointerException: Attempt to invoke virtual method 'android.content.Intent android.app.Activity.getIntent()' on a null object reference
07-03 17:12:13.878: I/Unity(32438):   at UnityEngine.AndroidJNISafe.CheckException () [0x00000] in <filename unknown>:0 
07-03 17:12:13.878: I/Unity(32438):   at UnityEngine.AndroidJNISafe.CallStaticVoidMethod (IntPtr clazz, IntPtr methodID, UnityEngine.jvalue[] args) [0x00000] in <filename unknown>:0 
07-03 17:12:13.878: I/Unity(32438):   at UnityEngine.AndroidJavaObject._CallStatic (System.String methodName, System.Object[] args) [0x00000] in <filename unknown>:0 
07-03 17:12:13.878: I/Unity(32438):   at UnityEngine.AndroidJavaObject.CallStatic (System.String methodName, System.Object[] args) [0x00000] in <filename unknown>:0 
07-03 17:12:13.878: I/Unity(32438):   at BranchAndroidWrapper.initSessionWithCallback (System.String callbackId) [0x00000] in <filename unknown>:0 
07-03 17:12:13.878: I/Unity(32438):   at Branch._initSessionWithCallback (System.String callbackId) [0x00000] in <filename unknown>:0 
07-03 17:12:13.878: I/Unity(32438):   at Branch.initSession (.BranchCallbackWithParams callback) [0x00000] in <filename unknown>:0 
07-03 17:12:13.878: I/Unity(32438):   at Main.Start () [0x00000] in <filename unknown>:
```

However opening the game without the link (no parameters), gives no problem and Branch is correctly initiated. As I showed you last time, the Init code is just this:

```
 try
            {
                Branch.initSession(
                    (parameters, error) =>
                    {
                        //Show error or correct initialization.
                        if (error != null)
                        {
                            Debug.LogError("Branch Error: " + error);
                        }
                        else if (parameters.Count > 0)
                        {
                            Debug.LogWarning("Branch initialization completed with the following parameters: " + parameters.Keys);
                        }

                        //Check if branch parameters (SHOULD BE CALLED HERE BECAUSE THIS CALLBACK IS EXECUTED IN THE UI THREAD).
                        Dictionary<string, object> lastParameters = Branch.getLatestReferringParams();
                        if (lastParameters != null && lastParameters.Count > 0)
                        {
                            //Debug parameters.
                            Debug.LogWarning("Received " + lastParameters.Count + " Branch parameters.");
                            foreach (object value in lastParameters.Values)
                            {
                                Debug.Log("    " + value.ToString());
                            }
                            //Pass to manager.
                            invitationAndMessagesManager.handleMobileInvitation(lastParameters);
                        }
                    });
            }
            catch (Exception e)
            {
                Debug.LogError("Error starting Branch!");
                Debug.LogError(e.Message);
            }
```

And the sent parameters in the URL should be these:

Dictionary<string, object> parameters = new Dictionary<string, object>
            {
                {"fullname", fullname},
                {"mode", "ROOM_INVITATION"},
                {"roomId", roomId},
                {"timestamp", timestamp},
                {"msgId", "" + UnityEngine.Random.Range(0, int.MaxValue)}
            };

Maybe is there something more that we have to do when initiating with URL parameters?

**Comment by aaustin** [comment]
Hmm, it's hard to tell immediately, but it seems that perhaps the Unity Activity is null. Which Activity does your intent filter point to?

**Comment by aaustin** [comment]
It's pointing to default activity.

   <application android:icon="@drawable/app_icon" android:label="@string/app_name" android:debuggable="true">
   <activity android:name="com.unity3d.player.UnityPlayerNativeActivity" android:label="@string/app_name">

```
<intent-filter>
<action android:name="android.intent.action.MAIN" />
<category android:name="android.intent.category.LAUNCHER" />
<category android:name="android.intent.category.LEANBACK_LAUNCHER" />
</intent-filter>

<!-- Branch -->
<intent-filter>
<data android:scheme="Bosonauts" android:host="open" />
<action android:name="android.intent.action.VIEW" />
<category android:name="android.intent.category.DEFAULT" />
<category android:name="android.intent.category.BROWSABLE" />
</intent-filter>

<meta-data android:name="unityplayer.UnityActivity" android:value="true" />
<meta-data android:name="unityplayer.ForwardNativeEventsToDalvik" android:value="false" />
```

</activity>

**Comment by jemchicomac** [comment]
Hi Austin, we waiting for this issue to be solved.

When do you think this can be solved?

**Comment by aaustin** [comment]
I'm really sorry on the delay @jemchicomac. We heard a report that it should be resolved tomorrow. I'll let you know as soon as I hear back for sure.

**Comment by shortstuffsushi** [comment]
Hi guys, this should be fixed by #3. Could you please review with the new version, and reopen if this is still an issue?
